src/java/com/aap/gitst/Logger_copy.java

```diff
@@ -0,0 +1,340 @@
+            pbar.start();
+            return pbar;
+        }
+    }
+
+    public boolean isDebugEnabled() {
+        return getLevel() == Level.DEBUG;
+    }
+
+    public boolean isInfoEnabled() {
+        return getLevel().ordinal() <= Level.INFO.ordinal();
+    }
+
+    public boolean isWarnEnabled() {
+        return getLevel().ordinal() <= Level.WARNING.ordinal();
+    }
+
+    public boolean isErrorEnabled() {
+        return getLevel().ordinal() <= Level.ERROR.ordinal();
+    }
+
+    private synchronized void log(final Object msg) {
+        if (!_pbar.isEmpty()) {
+            clearPogress();
+            _logWriter.println(msg);
+            printProgress();
+        } else {
+            _logWriter.println(msg);
+        }
+
+        _logWriter.flush();
+    }
+
+    // @GuardedBy("this")
+    private void clearPogress() {
+        _logWriter.print('\r');
+        for (final PBar b : _pbar) {
+            for (int i = 0; i < b._lastMessageLen; i++) {
+                _logWriter.print(' ');
+            }
+            _logWriter.print(' ');
+        }
+        _logWriter.print('\r');
+    }
+
+    // @GuardedBy("this")
+    private void printProgress() {
+        for (final PBar b : _pbar) {
+            b.printMessage();
+            _logWriter.print(' ');
+        }
+        _logWriter.flush();
+    }
+
+    public static enum Level {
+        DEBUG,
+        INFO,
+        WARNING,
+        ERROR,
+        OFF;
+    }
+
+    public static interface ProgressBar {
+
+        public void done(int count);
+
+        public void update();
+
+        public void complete();
+
+        public void close();
+    }
+
+    public static class DummyProgressBar implements ProgressBar {
+
+        @Override
+        public void done(final int count) {
+        }
+
+        @Override
+        public void update() {
+        }
+
+        @Override
+        public void complete() {
+        }
+
+        @Override
+        public void close() {
+        }
+    }
+
+    public class PBar extends Thread implements ProgressBar {
+        private final Object _message;
+        private final int _total;
+        private final AtomicInteger _counter;
+        private volatile boolean _closeOnComplete;
+        private volatile boolean _update;
+        private int _lastMessageLen;
+
+        public PBar(final Object message, final int total,
+                final boolean closeOnComplete) {
+            _message = message;
+            _total = total;
+            _closeOnComplete = closeOnComplete;
+            _counter = new AtomicInteger(total);
+        }
+
+        @Override
+        public void run() {
+            for (int prev = 0; !isInterrupted();) {
+                final int current = _counter.get();
+
+                if (current <= 0) {
+                    break;
+                } else if (!_update && (current == prev)) {
+                    LockSupport.park();
+                } else {
+                    prev = current;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)